web/modules/custom/joinup_invite/src/InvitationMessageHelper.php

```diff
@@ -93,6 +93,9 @@ protected function getDefaultArguments(InvitationInterface $invitation): array {
       $arguments["@invitation:${action}_url"] = Url::fromRoute('joinup_invite.update_invitation', $url_arguments, $url_options)->toString();
     }
 
+    $arguments['@invitation:role'] = $invitation->field_invitation_og_role->isEmpty() ? '' :
+      strtolower($invitation->field_invitation_og_role->first()->entity->label());
+
```

Comment:
Checking for the field being a list of FieldItemListInterface is like checking that the field exists.. If it does not exist, as it should, it is only normal that the site should break because something is quite wrong.
The `::isEmpty()` as far as I understand returns false if there isn't any value. This means that if it is not empty, at least one value exists so `::first()` is also safe to call. 
Last, the `->entity` will not break if the `target_id` property is null. It will simply return null.
So I narrowed down the check to two cases. Please, check the new version and let me know.

Reply:
> If it does not exist, as it should, it is only normal that the site should break because something is quite wrong.

We should not assume that fields are present on entities. They are optional in Drupal, meaning that this value can be `NULL` depending on the bundle type that is passed in. Ideally we should stop using the Field API directly and only use bundle classes where we _can_ be sure that a certain field is defined on the bundle.

For example if we call `InvitationMessageHelper::createMessage()` and pass in an invitation entity for a discussion then this will throw a fatal error. The `field_invitation_og_role` field is only defined on the `group_membership` invitation bundle and not on the `discussion` invitation bundle. So we need to check that it exists. We can also add more bundles in the future to invite people to other things, and they may or may not have this field.

But even if all the bundles would have this field then it is still dangerous to assume that a certain field will be populated under all circumstances. We had several occasions of fatal errors because of this recently, and one of which was because of a very similar line of code (ref https://github.com/ec-europa/joinup-dev/pull/2190#issuecomment-656281749)

```
$collection = $community_content->get('og_audience')->entity;
```

In this case the developer also assumed that all community content is always associated with a collection, so there always is an `og_audience` field with an entity attached. This sounds very reasonable, but it turned out to be not always true, and we had a fatal error in production when a moderator deleted a collection that contained pinned content.

Here are some examples of cases when expected field data can be missing, and it might be `NULL`:
* During site install and database updates.
* Acting on a stale entity that is coming from a static cache.
* Code being called during entity delete hooks, when references are already orphaned.
* Code running in a shutdown hook after the original data has already been deleted, e.g. during `@afterScenario` hooks in Behat tests.
* Cron jobs that run at a point in the future when relevant data has in the meanwhile been changed or deleted.